src/imagery/i.eodag/i.eodag.py

```diff
@@ -0,0 +1,326 @@
+#!/usr/bin/env python3
+
+############################################################################
+#
+# MODULE:      i.eodag
+#
+# AUTHOR(S):   Hamed Elgizery
+# MENTOR(S):   Luca Delucchi, Veronica Andreo, Stefan Blumentrath
+#
+# PURPOSE:     Downloads imagery datasets e.g. Landsat, Sentinel, and MODIS
+#              using EODAG API.
+# COPYRIGHT:   (C) 2024-2025 by Hamed Elgizery, and the GRASS development team
+#
+#              This program is free software under the GNU General Public
+#              License (>=v2). Read the file COPYING that comes with GRASS
+#              for details.
+#
+#############################################################################
+
+
+# %Module
+# % description: Eodag interface to install imagery datasets from various providers.
+# % keyword: imagery
+# % keyword: eodag
+# % keyword: sentinel
+# % keyword: landsat
+# % keyword: modis
+# % keyword: datasets
+# % keyword: download
+# %end
+
+# %option
+# % key: dataset
+# % type: string
+# % description: Imagery dataset to search for
+# % required: yes
+# % answer: S1_SAR_GRD
+# % guisection: Filter
+# %end
+
+# %option G_OPT_M_DIR
+# % key: output
+# % description: Name for output directory where to store downloaded data OR search results
+# % required: no
+# % guisection: Output
+# %end
+
+# %option
+# % key: id
+# % type: string
```

Comment:
Is it possible users would want to download a lager number of files that way? Then we should consider also the possibility of "file" input.

Reply:
Handled further here: https://github.com/HamedElgizery/grass-addons/issues/6